Objects/dictobject.c

```diff
@@ -464,6 +464,12 @@ estimate_keysize(Py_ssize_t n)
         (d)->ma_keys->dk_lookup = lookdict_unicode; \
     }
 
+#define MAYBE_NONUNICODE(d, key) \
+    if (!PyUnicode_CheckExact(key) && (d)->ma_keys->dk_lookup != lookdict) { \
+        assert((d)->ma_keys->dk_lookup == lookdict_unicode || (d)->ma_keys->dk_lookup == lookdict_unicode_nodummy); \
```

Comment:
This seems redundant.
`(d)->ma_keys->dk_lookup != lookdict` implies `(d)->ma_keys->dk_lookup == lookdict_unicode || (d)->ma_keys->dk_lookup == lookdict_unicode_nodummy`

Reply:
I removed the assertion.